### PR TITLE
RavenDB-7549 

### PR DESCRIPTION
**Pull request by aviv86**

### CODE REVIEW

File: src/Raven.Server/Web/System/DatabasesHandler.cs [hunk 1]
```diff
@@ -195,6 +196,25 @@ private Task DbInfo(string dbName)
                     var promotableTask = new PromotableTask(promotable, url, databaseName);
                     nodesTopology.Promotables.Add(GetNodeId(node, topology.WhoseTaskIsIt(promotableTask, ServerStore.IsPassive())));
                 }
+                
+                if (ServerStore.Engine.LeaderTag == ServerStore.NodeTag)
```
Comment (ayende) [inline src/Raven.Server/Web/System/DatabasesHandler.cs]: Why are you doing that? This is going into persistent storage, and the data look ephemeral to me.

File: src/Raven.Client/Server/Operations/DatabasesInfo.cs [hunk 2]
```diff
@@ -119,4 +123,24 @@ public DynamicJsonValue ToJson()
             };
         }
     }
+
+    public class DbGroupNodeStatus : IDynamicJson
+    {
+        public string ConnectionStatus;
+        public string LastError;
+        public DateTime LastUpdateDateTime;
```
Comment (ml054) [inline src/Raven.Client/Server/Operations/DatabasesInfo.cs]: Should we use nullable here?

File: src/Raven.Server/Web/System/DatabasesHandler.cs (hunk 1 above)
Comment (ml054) [reply src/Raven.Server/Web/System/DatabasesHandler.cs]: @aviv86 It means this information will be only available when client connects to leader?

File: src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs [hunk 3]
```diff
@@ -183,6 +186,33 @@ class ClusterObserver : IDisposable
...
                         }
+
+                        string reason;
+                        if (nodeStats == null)
+                        {
+                            reason = "Demoted because it had no status report in the latest cluster stats";
+                        }
+                        else if (nodeStats.LastReportStatus != ClusterNodeStatusReport.ReportStatus.Ok)
+                        {
+                            reason = $"Demoted because the last report status was \"{nodeStats.LastReportStatus}\" ";
```
Comment (ayende) [inline src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs:197]: Need full error here.

File: src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs [hunk 4]
```diff
@@ -183,6 +186,33 @@ class ClusterObserver : IDisposable
...
+                        else if (nodeStats.LastReport.TryGetValue(dbName, out var stats) && stats.Status == DatabaseStatus.Faulted)
+                        {
+                            reason = "Demoted because the DatabaseStatus for this node is Faulted";
+                        }
+                        else
+                        {
+                            reason = "Demoted because it had no DatabaseStatusReport";
+                        }
+
+                        if (nodeStats?.LastError?.Message != null)
```
Comment (ayende) [inline src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs:208]: If there is an error, append all of it.

File: src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs [hunk 5]
```diff
@@ -183,6 +186,33 @@ class ClusterObserver : IDisposable
...
+                        {
+                            reason += $". {nodeStats.LastError.Message}";
+                        }
+
+                        topology.DemotionReasons[member] = reason;
```
Comment (ayende) [inline src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs:213]: Try to avoid allocations

**Review by ayende — CHANGES_REQUESTED** [review]

**Pull request merged**